apps/default/loki/config.yaml

```diff
@@ -13,30 +11,41 @@ common:
   ring:
     kvstore:
       store: inmemory
-
 schema_config:
   configs:
     - from: 2020-10-24
-      store: boltdb-shipper
+      store: tsdb
       object_store: filesystem
-      schema: v11
+      schema: v13
       index:
         prefix: index_
         period: 24h
-
 query_scheduler:
   max_outstanding_requests_per_tenant: 4096
 frontend:
   max_outstanding_per_tenant: 4096
 query_range:
   parallelise_shardable_queries: true
 limits_config:
-  split_queries_by_interval: 15m
-  max_query_parallelism: 32
+  allow_structured_metadata: true # This is required to allow Loki to handle OpenTelemetry data https://grafana.com/docs/loki/latest/send-data/otel/#loki-configuration
+  reject_old_samples: false
+  reject_old_samples_max_age: 365d
+  split_queries_by_interval: 15m # This setting dictates how many parallel queries will be ran. If you are often searching over large time spans having this too low will result in excessive parallel overhead which can work against you depending on resources of the container
+  max_query_length: 0 # This allows you to query data over any time range. Default is 30d1h
+  tsdb_max_query_parallelism: 32
+  otlp_config:
+    resource_attributes:
+      ignore_defaults: true # ignores any default resource attributes listed in default_resource_attributes_as_index_labels
+      attributes_config:
+        - action: index_label
+          attributes:
+            - environment.name
```

Comment:
Why this syntax over setting `default_resource_attributes_as_index_labels`?

Reply:
https://grafana.com/docs/loki/latest/send-data/otel/#changing-the-default-mapping-of-otlp-to-loki-format.

I preferred this syntax over `default_resource_attributes_as_index_labels` due to the fact that this syntax does support patterns using regex. Made it easier to see which resource attributes were being set as index labels in the beginning. This code snippet uses a list since `environment.name` was the only index label that was displayed in grafana. In inclusion you can also configure `otlp_config` to set scope and log attributes to `structured_metadata` or drop certain attributes. In this case since the attributes are set by default to be structured_metadata and didn't drop any attributes I didn't need to include it here.